File: SingletonDemo/Singleton.cs
class Logger {
  private StreamWriter writer;

  public Logger() {
    writer = new StreamWriter("app.log");
    Console.WriteLine(writer);
  }

  public void info(String message) {
    writer.WriteLine("INFO: " + message);
  }

  public void close() {
    writer.Close();
  }
}

public class Program {
  public static void Main(){
    var logger = new Logger();
    logger.info("This is information");
    logger.close();
  }
}

File: SingletonDemo/singleton.cpp
#include <iostream>
#include <fstream>
#include <string>

using namespace std;

class Logger {
    fstream writer;
    public:
    Logger() {
        writer.open("app.log", fstream::out | fstream::ate | fstream::app);
    }

    ~Logger() {
        this->close();
    }

    void info(string message) {
        writer << "INFO: " << message << endl;
    }

    void close() {
        writer.close();
    }
};

int main() {
    auto log = new Logger();
    log->info("This is information");
    log->close();
    return 0;
}
